badges fixes and cached counts

xithan opened the pull request:
didn't work properly on imports

xithan committed:
award fixed
didn't work properly on imports

xithan committed:
:dog2:

xithan committed:
typo

xithan committed:
:dog2:

xithan committed:
refactor pointer_spec

xithan committed:
make it green

xithan committed:
cached counts

ethn commented on mod/badges/set/abstract/award_badges.rb:
```diff
@@ -20,10 +20,14 @@ def earns_badge action
+    director = ActManager.fetch(badge_pointer)
+    director.reset_stage
+    badge_pointer = director.card
+  end
```
maybe separate into #badge_pointer method?

ethn replied:
why does fetch not find the correct card?  Isn't it supposed to?

I've been thinking that the ActManager should be pretty much invisible to mod developers in most cases.  Is that how you think about it?

xithan replied:
I'm not sure if `badge_pointer = director.card` is really necessary. I think I added that before I found the real issue which is that on metric value imports we add all metric values in the integrate stage of the import card. When the first metric value adds a badge to the badge_pointer the badge_pointer catches up to the integrate stage and all badges added later to the badge_pointer by other metric values of that import are not saved (because the storage phase for the badge_pointer is already completed)

ethn replied:
So do you think we can take out the ActManager part?

If it stays in, I think we should be able to explain the cases in which a mod developer would be expected to use this kind of thing, no?

xithan replied:
we can try to let `attach_subcard` handle that issue

xithan commented:
I pushed accidentally cached count changes to this PR

ethn commented on mod/metric_answer_lookup/lib/answer.rb:
```diff
@@ -74,22 +74,27 @@ def importance_sort mas, args
+        count += batch.size
+        puts "#{batch.first} - #{count}"
+        refresh(batch, *fields)
       end
```
much better!!

ethn commented on mod/metrics/spec/set/type/metric_value_import_file_spec.rb:
```diff
@@ -77,8 +78,41 @@ def run_import
+      expect(badge_names)
+        .to include "Apple Inc.+Researcher+company badge",
+                            "Samsung+Researcher+company badge"
+    end
```
+1

ethn commented on mod/wikirate/spec/set/type/pointer/pointer_spec.rb:
```diff
@@ -1,134 +1,122 @@
+    end
+    let(:elbert_punchline) do
+      create "Elbert Hubbard+punchline",
+             content: "You will never get out of it alive."
```
nice examples!

ethn commented on mod/wikirate/spec/set/type/pointer/pointer_spec.rb:
```diff
@@ -1,134 +1,122 @@
+                { name: "Elbert Hubbard",
+                  type: "Basic",
+                  content: "Do not take life too seriously." },
+                { name: "Elbert Hubbard+punchline",
```
I disagree with the hound here (and agree with you)

xithan replied:
Maybe it's time to say good bye to our hound. Hard to find the real comments between all those woofs. Codeclimate has improved a lot and does an at least equally good maybe even better job of marking style violations.

ethn reviewed (commented):
I would be interested in hearing why the ActManager stuff is necessary.  It's a clever fix, but I'm wondering if we can find a way to solve this problem by improving ActManager or caching or something.

(I'm guessing you will persuade me that's not practical yet and we will merge as is, but I wanted to ask anyway)

xithan committed:
more cached counts and specs clean up

xithan committed:
topic cached counts

xithan committed:
Merge branch 'master' of http://github.com/wagn/wikirate into bugfixes

xithan committed:
remove idea config

xithan committed:
fix some specs

xithan committed:
refactor cached counts

xithan committed:
green?

xithan committed:
wagn tick

xithan committed:
:dog2:

xithan committed:
add "or" step to fix double_check hover issue

xithan committed:
:dog2:

xithan committed:
update test seed db

xithan committed:
remove idea config files

ethn commented on db/migrate_cards/20170330102819_cached_count_table.rb:
```diff
@@ -0,0 +1,19 @@
+      t.integer :left_id
+      t.integer :right_id
+      t.integer :value
+    end
```
so simple.  thing of beauty.

ethn commented on db/migrate_cards/20170401182741_remove_cached_count_cards.rb:
```diff
@@ -0,0 +1,36 @@
+    remove type: :wikirate_topic, plus_right: :wikirate_company
+  end
+
+  def remove type: , plus_right:
```
what do these colons mean?

xithan replied:
required keyword arguments (https://robots.thoughtbot.com/ruby-2-keyword-arguments)

ethn commented on mod/badges/set/abstract/answer_create_affinity_badge.rb:
```diff
@@ -31,6 +31,14 @@ def virtual?
+end
+
+def badge
+  cardname.parts[1]
```
why does this file both delegate `badge` to card _and_ define badge?

xithan replied:
it delegates `badge` from format to card

ethn commented on mod/caching/spec/set/type_plus_right/metric/source_spec.rb:
```diff
@@ -0,0 +1,14 @@
+require_relative "../../../support/cached_count_shared_examples"
```
I like these shared examples!!

ethn commented on mod/wikirate/spec/set/right/wikirate_link_spec.rb:
```diff
@@ -8,7 +8,7 @@
   # it "should block changing the url for existing source" do
   #   # create the page with source
-  #   url = "http://www.google.com/?q=wikirateissocoolandawesomeyouknowsigh"
+  #   url = "http://www.google.com/?q=wikiratesigh"
```
haha.  I think this may have slightly changed the meaning.

xithan replied:
that was a global search and replace of `wikirateissocoolandawesomeyouknow` :smile:

ethn reviewed (approved):
great stuff.  Still a comment or two worth attention, but ready to roll

xithan committed:
view to temlate

xithan committed:
finalizing cached counts

xithan committed:
Merge branch 'master' of https://github.com/wagn/wikirate into bugfixes

xithan committed:
Merge branch 'bugfixes' of https://github.com/xithan/wikirate into bugfixes

xithan committed:
wagn tick

xithan committed:
idea config

(cherry picked from commit 930ec7b29f3ded4a58d9d9dcc71c2565e1458ae8)

xithan committed:
update Gemfile.lock

xithan committed:
wagn tick

xithan committed:
wagn tick

xithan committed:
wagn tick

xithan committed:
wagn tick

xithan committed:
bugfixes
found_by issue
haml template path
Death Star change on wikipedia

xithan committed:
wagn tick

xithan committed:
wagn tick

xithan committed:
add machine output files

xithan committed:
wagn tick

xithan committed:
mods tick

xithan committed:
speed up features
add vote counts to test db so that they don't have to be
generated during the tests

xithan committed:
fix limit bug

xithan committed:
try without waiting

xithan committed:
exclude test files in codeclimate

xithan committed:
be patient
